### PR TITLE
chore(*): fix warnings on non-linux platforms

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Fix the warnings introduced by #7305 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Review by st1page — APPROVED** [review]
LGTM

File: src/compute/src/memory_management/memory_manager.rs [hunk 1]
```diff
@@ -12,20 +12,24 @@
...
 
 use risingwave_batch::task::BatchManager;
+#[cfg(target_os = "linux")]
 use risingwave_common::util::epoch::Epoch;
 use risingwave_stream::executor::monitor::StreamingMetrics;
 use risingwave_stream::task::LocalStreamManager;
 #[cfg(target_os = "linux")]
 use tikv_jemalloc_ctl::{epoch as jemalloc_epoch, stats as jemalloc_stats};
+#[cfg(target_os = "linux")]
 use tracing;
```
Comment (ice1000) [inline src/compute/src/memory_management/memory_manager.rs:28]: Why not make these local to the linux-only function?

**Comment by ice1000** [comment]
image


This is embarrasing. I'll make another PR

**Review by yuhao-su — APPROVED** [review]

**Comment by TennyZhuang** [comment]
@ice1000 When can you submit your PR? The warnings confused me so I always ignored the real warnings. If it's OK, we can merge the PR first.

**Review by BugenZhao — APPROVED** [review]

**Review by BowenXiao1999 — APPROVED** [review]
Agree. Indeed it impose some bad experiences for local dev.

**Comment by ice1000** [comment]
> @ice1000 When can you submit your PR? The warnings confused me so I always ignored the real warnings. If it's OK, we can merge the PR first.

Sure

**Review by ice1000 — APPROVED** [review]

**Pull request merged**